### ISSUE TITLE
Implement Fast Hyperparameter Tuning Pipeline for GP Training

### ISSUE THREAD

**Issue by Klinjin** [Klinjin/DH_profile_kSZ_WL#5]
## Problem Statement

GP training on full dataset (69,632 samples, 21 radius bins) takes **3-5 days**, making hyperparameter tuning impractical. Testing different kernels and learning rates would take weeks.

### New Features Added
- `tune_hyperparameters()` method in `GPTrainer` class
- `train_single_gp_model_with_validation()` with early stopping
- Comprehensive result logging and time estimation

### Key Optimizations
- **Subset training**: Uses 10% of data (6,963 samples) instead of full dataset
- **Limited radius bins**: Tests only first 5 bins instead of all 21
- **Early stopping**: Validation-based stopping with configurable patience
- **Time estimation**: Predicts full training time from subset results

## Performance Impact

| Metric | Before | After |
|--------|--------|--------|
| Hyperparameter search | Weeks | **10-30 minutes** |
| Full training time | Unknown | 3-5 days (estimated) |
| Configurations tested | Manual 1-by-1 | **12 configs automatically** |

## Usage Example

```python
# Fast hyperparameter search (10-30 minutes)
tuning_results = trainer.tune_hyperparameters(
    subset_ratio=0.1,  # 10% of data for speed
    lr_candidates=[1e-4, 3e-4, 1e-3, 3e-3],
    kernel_types=['hierarchical', 'robust', 'physics_informed'], 
    max_iter_tune=500,
    early_stop_patience=50,
    n_radius_bins_tune=5  # Only tune on first 5 bins
)

# Get best config and train full model
best_config = tuning_results['best_config'] 
trainer.train(
    kernel_type=best_config['kernel_type'], 
    lr=best_config['learning_rate'],
    maxiter=5000  # Full iterations for final training
)
```

## Files Modified
- `src/models/gp_trainer.py`: Added `tune_hyperparameters()` method
- `src/models/gp_trainer_one.py`: Added `train_single_gp_model_with_validation()`

## Next Steps
- [x] Test hyperparameter tuning pipeline on real data
- [ ] Integrate sparse GP methods for further speedup (inducing points)
- [ ] Implement parallel training across radius bins
- [ ] Compare subset-based hyperparameter selection to full training results

## Scientific Impact
This enables practical hyperparameter optimization for cosmological parameter inference, allowing us to:
1. Find optimal GP configurations efficiently
2. Reduce overfitting through proper validation
3. Scale to larger datasets confidently
4. Accelerate the NPE integration workflow (Step 5)

**Estimated Time Savings**: Weeks → 30 minutes for hyperparameter search
